dattri/algorithm/influence_function.py

```diff
@@ -201,59 +209,416 @@ def lissa_collate_fn(
+                parameters.
+            data (Tuple[Tensor]): The test data. Normally this is a tuple
                 of input data and target data, the number of items in the
                 tuple should be aligned in the target function. The tensors'
                 shape follows (1, batchsize, ...).
-            query (torch.Tensor): The query to be transformed. Normally it is
+
+        Returns:
+            torch.Tensor: The query based on the test data. Normally it is
                 a 2-d dimensional tensor with the shape of
                 (batchsize, num_parameters).
-            transformation_kwargs (Dict[str, Any]): The keyword arguments for
-                the transformation function.
+        """
+        model_params, _ = self.task.get_param(index)
+        return self.task.get_grad_target_func()(model_params, data)
+
+    def generate_train_query(
+        self,
+        index: int,
+        data: Tuple[torch.Tensor, ...],
+    ) -> torch.Tensor:
+        """Calculating the query based on the train data.
+
+        Inner product attributor calculates the inner product between the
+        train query and the transformation of the test query. This function
+        calculates the train query based on the train data.
+
+        Args:
+            index (int): The index of the model parameters. This index
+                is used for ensembling of different trained model
+                parameters.
+            data (Tuple[Tensor]): The train data. Normally this is a tuple
+                of input data and target data, the number of items in the
+                tuple should be aligned in the target function. The tensors'
+                shape follows (1, batchsize, ...).
 
         Returns:
-            torch.Tensor: The transformation on the query. Normally it is a 2-d
-                dimensional tensor with the shape of (batchsize, transformed_dimension).
+            torch.Tensor: The query based on the train data. Normally it is
+                a 2-d dimensional tensor with the shape of
+                (batchsize, num_parameters).
         """
-        from dattri.func.fisher import ifvp_datainf
+        model_params, _ = self.task.get_param(index)
+        return self.task.get_grad_loss_func()(model_params, data)
 
-        model_params, param_layer_map = self.task.get_param(index, layer_split=True)
+    def cache(self, full_train_dataloader: DataLoader) -> None:
+        """Cache the dataset for inverse hessian calculation.
 
-        self.ihvp_func = ifvp_datainf(
-            self.task.get_loss_func(),
-            0,
-            (None, 0),
-            param_layer_map=param_layer_map,
-            **transformation_kwargs,
+        Args:
+            full_train_dataloader (DataLoader): The dataloader
+                with full training samples for inverse hessian calculation.
+        """
+        self._set_full_train_data(full_train_dataloader)
+
+    @staticmethod
```

Comment:
Why staticmethod?

Reply:
Because the ihvp method does not alter anything of the class itself and at the same time I want to make it a class function.